source/common/http/codec_helper.h

```diff
@@ -25,6 +34,7 @@ class StreamCallbackHelper {
       return;
     }
 
+    reset_callbacks_started_ = true;
```

Comment:
I think one of the variables still needs to be deleted? Will keep reviewing otherwise.